src/main/java/htsjdk/samtools/reference/AbstractFastaSequenceFile.java

```diff
@@ -82,32 +73,41 @@
     AbstractFastaSequenceFile(final Path path, final String source, final SAMSequenceDictionary sequenceDictionary) {
         this.path = path;
         this.source = source;
-        this.sequenceDictionary = sequenceDictionary;
+        this.dictionary = new Lazy<>(() -> sequenceDictionary);
     }
 
-    protected static File findSequenceDictionary(final File file) {
-        final Path dictionary = findSequenceDictionary(IOUtil.toPath(file));
-        if (dictionary == null) {
-            return null;
+    /** Attempts to find and load the sequence dictionary if present. */
+    protected SAMSequenceDictionary findAndLoadSequenceDictionary(final Path fasta) {
+        final Path dictPath = findSequenceDictionary(path);
+        if (dictPath == null) return null;
+
+        IOUtil.assertFileIsReadable(dictPath);
+        try (InputStream dictionaryIn = IOUtil.openFileForReading(dictPath)) {
```

Comment:
Is this change to support dict.gz files?  Should findSequenceDictionary below try to locate them as well?

Reply:
Not really, although I suppose it now could.  I just like for every file/path open to vector through IOUtil since it standardizes how things are done and how buffering is applied.